# Try to fix pointer assigned to int and vice-versa problems.  Written by Joe Pelkey.

--- a/actkbd.h
+++ b/actkbd.h
@@ -122,7 +122,7 @@
 typedef struct _attr_t attr_t;
 struct _attr_t {
     int type;			/* Attribute type */
-    void *opt;			/* Options for this attribute */
+    int opt;			/* Options for this attribute */
     attr_t *next;		/* The next node */
 };
 
--- a/config.c
+++ b/config.c
@@ -107,7 +107,7 @@
     strtolower(attrs);
     while ((tmp = strsep(&attrs, ", \t")) != NULL) {
 	int type = -1;
-	void *opt = NULL;
+	int opt = 0;
 	char *num = NULL;
 
 	if (strlen(tmp) == 0)
@@ -174,9 +174,9 @@
 
 	    errno = 0;
 	    if (strlen(num) > 0) {
-		opt = (void *)((int)strtol(num, (char **)NULL, 10));
+		opt = ((int)strtol(num, (char **)NULL, 10));
 	    } else {
-		opt = (void *)((int)(-1));
+		opt = ((int)(-1));
 	    }
 
 	    if (((int)opt < 0) &&
